### ISSUE TITLE
Hidden link text on Check answers pattern omit space character in JAWS

### ISSUE THREAD

**Issue by timpaul** [alphagov/govuk-frontend#2547]
The ‘Change’ links have hidden text, like this:

```
<a class="govuk-link" href="#">
Change<span class="govuk-visually-hidden"> name</span>
</a>
```
But apparently JAWS sometimes ignores the opening space in the hidden text, so it would read out `Changename` in the above example.

We think one fix might be to use a non-breaking space character instead, so it'd look like this:

```
<a class="govuk-link" href="#">
Change<span class="govuk-visually-hidden">&nbsp;name</span>
</a>
```

**Comment by timpaul** [comment]
Thanks to Craig Abbot for reporting this issue.

**Comment by abbott567** [comment]
Thanks @timpaul - it was originally raised by @htmlandbacon: [GOVUK Design System issue 1382](https://github.com/alphagov/govuk-design-system/issues/1382)

The reason for raising it again it because both HMRC and DWP have published guidance which uses a different mechanism for change links. It involves duplicating the content and using aria-hidden on one. For example:

```html
<span aria-hidden="true">Change</span>
<span class="govuk-visually-hidden">Change Elizabeth Kean</span>
```

If the CSS were to not load, then the visible link on the screen would be: `ChangeChange Elizabeth Kean`, which is weird, but my main concern is that we have a separation of guidance from the GOVUK Design System, and I worry will be confusing for teams.

Links:
- [DWP Design System guidance](https://design-system.dwp.gov.uk/patterns/add-another-thing#accessibility)
- [HMRC Design System guidance](https://design.tax.service.gov.uk/hmrc-design-patterns/add-to-a-list/)

**Comment by adamliptrot-oc** [comment]
I re-tested this in Feb 2021 I couldn’t hear the merging of content anymore in JAWS (tested 2019, 2020, 2021), although the merging is still evident in the links list dialog (though again not audible).

@abbott567 The original solution for the split copy included the use of a `hidden` attribute alongside the `aria`  to prevent the dual inclusion when CSS did not load, which was then overridden by CSS. Looks like this didn't make it into the pattern.

**Comment by adamliptrot-oc** [comment]
Just checked and NVDA is also still removing the space between the visible and hidden copy in the dialogs, although again I cannot hear the words merging.

If we do find that the words are being audibly merged then another option was suggested by Andrew Arch Principal Accessibility Consultant at Intopia on the Govt Accessibility email list. This is using a capital letter for the first word of the hidden copy which might result in the non-honouring of the space to be negated by the screen-reader pronunciation of camelcased text eg ChangeName where the screen-reader will read out the words as individual words due to the casing.

**Comment by querkmachine** [comment]
We've started getting more reports of this happening lately, including in screen readers other than JAWS (at least NVDA, VoiceOver and TalkBack). We've also noticed this occurring whilst working on other components [like the Exit this Page button](https://github.com/alphagov/govuk-frontend/pull/3268#discussion_r1102645838).

These issues seem to happen regardless of if the space is inside or outside of the visually-hidden span.

We may want to investigate a more robust method of doing this in future, such as some of the suggestions made above. This could potentially be done in conjunction with removing the hardcoded visually-hidden span in summary list links (#2649).

**Comment by owenatgov** [comment]
Has this been solved by https://github.com/alphagov/govuk-frontend/pull/3836?

**Comment by owenatgov** [comment]
Have repeated Hanna's testing from #3836 and I'm confident we've solved this.
